### PR TITLE
Add instanceOf propType

### PR DESCRIPTION
**Pull request by philpee2**
#MINOR#

### CODE REVIEW

File: addon/utils/prop-types.js [hunk 1]
```diff
@@ -134,5 +145,11 @@ PropTypes.oneOf = function (typeDefs) {
   return type
 }
 
+PropTypes.instanceOf = function (typeDef) {
```
Comment (sandersky) [inline addon/utils/prop-types.js]: This is overriding what is set by https://github.com/philpee2/ember-prop-types/blob/7b647c5ad5afe1267779d0992c5086af7152d5af/addon/utils/prop-types.js#L111. If you want to manually set this then you should not put `instanceOf` in the validators object at the top of the function and should instead do something similar to https://github.com/philpee2/ember-prop-types/blob/7b647c5ad5afe1267779d0992c5086af7152d5af/addon/utils/prop-types.js#L114.

**Comment by sandersky** [comment]
The build failed because it did not pass linting. Please remove semi-colons as we are following http://standardjs.com/

File: addon/utils/prop-types.js [hunk 2]
```diff
@@ -128,11 +128,28 @@ validators.oneOf = function (ctx, name, value, def) {
...
   const type = generateType('oneOf')
   type.isRequired.typeDefs = type.typeDefs = typeDefs
   return type
 }
 
+PropTypes.instanceOf = function (typeDef) {
+  const type = generateType('instanceOf')
+  type.isRequired.typeDef = type.typeDef = typeDef
+  return type
+};
```
Comment (sandersky) [inline addon/utils/prop-types.js]: You missed a semi-colon

**Comment by philpee2** [comment]
Got it. Removed the semicolons.

**Comment by sandersky** [comment]
:+1:

**Pull request merged**